lib/internal/modules/esm/loader.js

```diff
@@ -96,6 +100,59 @@ const {
 
 let emittedSpecifierResolutionWarning = false;
 
+function nextHookFactory(chain, meta, validate) {
+  // First, prepare the current
+  const { hookName } = meta;
+  const {
+    fn: hook,
+    url: hookFilePath,
+  } = chain[meta.hookIndex];
+
+  const nextHookName = `next${
+    StringPrototypeToUpperCase(hookName[0]) +
+    StringPrototypeSlice(hookName, 1)
+  }`;
+
+  // When hookIndex is 0, it's reached the default, which does not call next()
+  // so feed it a noop that blows up if called, so the problem is obvious.
+  const generatedHookIndex = meta.hookIndex;
+  let nextNextHook;
+  if (meta.hookIndex > 0) {
+    // Now, prepare the next: decrement the pointer so the next call to the
+    // factory generates the next link in the chain.
+    meta.hookIndex--;
+
+    nextNextHook = nextHookFactory(chain, meta, validate);
+  } else {
+    // eslint-disable-next-line func-name-matching
+    nextNextHook = function chainAdvancedTooFar() {
+      throw new RangeError(
+        `ESM custom loader '${hookName}' advanced beyond the end of the chain; this is a bug in Node.js itself.`
+      );
+    };
+  }
+
+  return ObjectDefineProperty(
+    async (...args) => {
+      // Update only when hook is invoked to avoid fingering the wrong filePath
+      meta.hookErrIdentifier = `${hookFilePath} '${hookName}'`;
+
+      validate(meta.hookErrIdentifier, ...args);
+
+      // Set when next<HookName> is actually called, not just generated.
+      if (generatedHookIndex === 0) { meta.chainFinished = true; }
+
+      const output = await hook(...args, nextNextHook);
+
+      if (output?.shortCircuit === true) { meta.shortCircuited = true; }
+
+      return output;
+    },
+    'name',
+    { value: nextHookName },
```

Comment:
https://github.com/nodejs/node/blob/df5664440e447c2724003700b5a81da8fadc4f5a/doc/contributing/primordials.md#L653-L678
```suggestion
    { __proto__: null, value: nextHookName },
```

Reply:
Done. But, I think this isn't terribly important? Setting the `nextHookName` is only for the user, so if the user chooses to eff this up 🤷‍♂️

Reply:
It’s important because the code will throw an error if `Object.prototype.get` is assigned to anything.